components/codec/src/byte.rs

```diff
@@ -98,6 +98,53 @@ impl MemComparableByteCodec {
         }
     }
 
+    /// Encodes all bytes in the `src` in ascending memory-comparable format in place.
+    /// If the capacity of `src` is less than the encoded length, `src` will `reserve` the
+    /// minimum sufficient capacity.
+    ///
+    /// Returns the number of bytes encoded.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the encoded length overflows usize.
+    pub fn encode_all_in_place(src: &mut [u8]) -> usize {
+        // Refer: https://github.com/facebook/mysql-5.6/wiki/MyRocks-record-format#memcomparable-format
+        let src_len = src.len();
+        let dest_len = Self::encoded_len(src_len);
```

Comment:
please add assertions to ensure that src has at least dest_len space. Otherwise this function may lead to memory corruptions.